chalice/app.py

```diff
@@ -379,6 +429,20 @@ def _register_view(view_func):
             return view_func
         return _register_view
 
+    def _validate_route_content_types(self, name, content_types):
+        # A routes content_types be homogeneous in their binary support.
+        binary, non_binary = [], []
+        for content_type in content_types:
+            if content_type in self.api.binary_types:
+                binary.append(content_type)
+            else:
+                non_binary.append(content_type)
+        if binary and non_binary:
+            raise ValueError(
+                'In view function "%s", the content_types %s support binary '
+                'and %s do not. All content_types must be consistent in their '
```

Comment:
So that doesn't really fix the issue though. Because for `image/tff` if I send the `Accept: image/tff` header API gateway will send me back base64 encoded string instead of the bytes for that image. Whereas for all the other types it will send me. 

I'm not against forcing the `raw_body` to be `bytes`, I think that makes sense, but I think there is also a difference in behavior that is very subtle and difficult do debug. I would rather be explicit about it.